src/sys/unix/selector/kqueue.rs

```diff
@@ -34,17 +34,6 @@ type Flags = u16;
 #[cfg(target_os = "netbsd")]
 type Flags = u32;
 
-// Type of the `data` field in the `kevent` structure.
-#[cfg(any(
-    target_os = "dragonfly",
-    target_os = "freebsd",
-    target_os = "ios",
-    target_os = "macos"
-))]
-type Data = libc::intptr_t;
```

Comment:
DragonFlyBSD and MacOS (and likely iOS) still use `intptr_t` in the manual. I think to be safe we should stick to that as well. You can just move FreeBSD down to NetBSD and OpenBSD.

Reply:
Except that FreeBSD 11 also still uses `intptr_t`.  And currently there's no way for a crate to select which libc ABI it wants to use.  So by removing that type entirely we'll just rely on type inference.

Reply:
But you're using `as i64` in multiple places below, so it doesn't use type interference.

Reply:
Yeah.  I changed the argument type of `kevent_register` to use i64 everywhere, instead of being platform-dependent.  So now there's only one place that needs type inference: line 284 (and line 61, before my second commit).  Those are the only places that actually use the libc type.